makefile.shade

```diff
@@ -1348,39 +1328,96 @@ functions @{
-        var argument = string.Format("list {0} --framework {1} --assemblies", projectFolder, framework);
-
-        if (coreclrFolder != null)
+        var runtimeTarget = "";
+        var args = "--runtime {0}";
+        if(target.OS == "win")
         {
-            argument += (" --runtime " + coreclrFolder);
+            runtimeTarget = "win7";
         }
-
-        int result;
-        if (!IsMono)
+        else if(target.OS == "linux")
         {
-            result = ExecuteAndRedirectOutput("cmd", string.Format("/C dnu {0}", argument), tmpFile);
+            runtimeTarget = "ubuntu.14.04";
         }
-        else 
+        else if(target.OS == "darwin")
         {
-            result = ExecuteAndRedirectOutput("dnu", argument, tmpFile);
+            runtimeTarget = "osx.10.10";
         }
 
-        if (result != 0)
+        runtimeTarget += "-" + target.Arch;
+        args = string.Format(args, runtimeTarget);
+
+        if (target.Flavor == "mono")
         {
-            throw new InvalidOperationException(string.Format("Return value for the command listing assemblies for project under {0} is not zero ({1}). Results are in temp file {2}.", projectFolder, result, tmpFile));
+            args = "";
         }
 
-        try
+        string tmp;
+        var home = "USERPROFILE";
+        if (IsMono)
+        {
+            ExecuteAndRedirectOutput("dnu", string.Format("restore {0} {1}", projectFolder, args), out tmp);
+            home = "HOME";
+        }
+        else
         {
-            return new HashSet<string>(File.ReadAllLines(tmpFile).Where(l => File.Exists(l)));
+            ExecuteAndRedirectOutput("cmd", string.Format("/C dnu restore {0} {1}", projectFolder, args), out tmp);
         }
-        catch (Exception ex)
+        
+        if (target.Flavor == "mono")
+        {
+            runtimeTarget = "DNX,Version=v4.5.1";
+        }
+        else
         {
-            throw new InvalidOperationException(string.Format("Unexpected exception thrown while parsing assemblies list for project under {0}. ({1})", projectFolder, ex.Message));
+            runtimeTarget = (target.Flavor == "coreclr" ? "DNXCore,Version=v5.0/" : "DNX,Version=v4.5.1/") + runtimeTarget;
+        }
+
+        var lockFile = File.ReadAllText(Path.Combine(projectFolder, "project.lock.json"));
+        var obj = (JsonObject)Json.Deserialize(lockFile);
+        var runtimePackages = obj.ValueAsJsonObject("targets").ValueAsJsonObject(runtimeTarget);
+        var projects = obj.ValueAsJsonObject("libraries");
+        
+        var projectAssemblies = new HashSet<string>();
+        foreach (var package in runtimePackages.Keys)
+        {
+            var lib = runtimePackages.ValueAsJsonObject(package);
+            if(lib.ValueAsString("type") == "project")
+            {
+                var name = package.Substring(0, package.IndexOf('/'));
+                if (name.EndsWith(".Sources") || name.Contains("Microsoft.Dnx.Runtime.Internals"))
```

Comment:
We should probably rename `Microsoft.Dnx.Runtime.Internals` to `Microsoft.Dnx.Runtime.Internals.Sources` to be consistent. Not something that should be done in this PR, just noting. /cc @davidfowl